plugin-influxdb/src/lib.rs

```diff
@@ -174,22 +161,60 @@ impl Output for InfluxDbOutput {
     }
 }
 
+// Returns true if the attribute with this key should be serialized as an InfluxDB tag,
+// false if it should become a field.
+fn partition_tag(
+    key: &str,
+    attributes_as: &AttributeAs,
+    attributes_as_tags: &HashSet<String>,
+    attributes_as_fields: &HashSet<String>,
+) -> bool {
+    match attributes_as {
+        AttributeAs::Tag => {
+            // default is tag => tag unless if in set
+            !attributes_as_fields.contains(key)
+        }
+        AttributeAs::Field => {
+            // default is field => tag only if in set
+            attributes_as_tags.contains(key)
+        }
+    }
+}
+
+// Check if the tag key is reserved and in case it is append a prefix
+fn ensure_reserved_tag_key(reserved_tags: [&str; 4], tag_key: String) -> String {
+    if reserved_tags.contains(&tag_key.as_str()) {
+        format!("alumet_attribute__{tag_key}")
+    } else {
+        tag_key
+    }
+}
+
+// Check if the field key is reserved and in case it is return a predefined key
+fn ensure_reserved_field_key<'a>(reserved_field: &'a str, field_key: &'a str) -> &'a str {
+    if field_key == reserved_field {
+        "alumet_attribute__value"
+    } else {
+        field_key
+    }
+}
+
 #[derive(Serialize, Deserialize)]
 #[serde(deny_unknown_fields)]
-struct Config {
-    host: String,
-    token: String,
-    org: String,
-    bucket: String,
-    attributes_as: AttributeAs,
-    attributes_as_tags: Option<HashSet<String>>,
-    attributes_as_fields: Option<HashSet<String>>,
+pub struct Config {
+    pub host: String,
+    pub token: String,
+    pub org: String,
+    pub bucket: String,
+    pub attributes_as: AttributeAs,
+    pub attributes_as_tags: Option<HashSet<String>>,
+    pub attributes_as_fields: Option<HashSet<String>>,
 }
 
 /// How to serialize Alumet attributes by default?
 #[derive(Serialize, Deserialize)]
```

Comment:
```suggestion
#[derive(Serialize, Deserialize, Clone, Copy)]
```
Making the `AttributeAs` enum `Copy` will simplify its use: you will be able to remove all the references everywhere and just pass it directly. It makes sense because the enum is very simple (only two plain values), so it will end up as one byte, so there's no need to pass references everywhere.

For instance you'll be able to do:
```rs
assert_eq!(partition_tag("is_a_tag", AttributeAs::Tag, &tags, &fields), true);
```